index.js

```diff
@@ -60,3 +64,44 @@ function checkOptions(options) {
     });
   });
 }
+
+/**
+ * Run through directives to see if any value
+ * is a function.
+ * @param {Object.<String,Array|String|Function>}
+ * @return {Boolean}
+ */
+function isDynamic(directives) {
+  return some(directives, function(directive) {
+    directive = [].concat(directive); // cast to array
+    return some(directive, function(val) {
+      return typeof val === "function";
+    });
+  });
+}
+
+/**
+ * Parses dynamic directives, returning a brand
+ * new object where all functions have been
+ * turned into string
+ * @param {Object.<String,Array|String|Function>
+ * @return {Object.<String,Array|String>}
+ */
+function parseDynamic(directives, args) {
+
+  // a recursive function to convert any dynamic values into static ones
+  function handle(value) {
```

Comment:
Does this function need to be redefined for every call to `parseDynamic`? Could we move it out of scope?